test/fuzz.jl

```diff
@@ -0,0 +1,178 @@
+using StyledStrings: styled
+using Supposition
```

Comment:
Add Supposition into the project's `[extras]` as a test target?

Reply:
I could, and do the same trick as Tar.jl does to only run the test in CI, however, it's not terribly quick to run (~30s), and only relevant when changes to `styledmarkup.jl` are made. This is why I've left it as just a big note of "run this if you edit this file" at the start of `styledmarkup.jl` for now.

I'm not sure what the best course of action is, this is just what's occurred to me.

Reply:
You can cut the execution time of a "everything passes" invocation of `@check` that's more-or-less only there to safeguard against regressions by setting a smaller  `max_examples`, from the default of `10_000`. E.g. if you set `max_examples = 1_000`, that should roughly cut execution time by a factor of ten. This shouldn't impact the quality of counterexamples that are found, since `max_examples` is just for steering the maximum of examples before Supposition.jl "gives up" and claims that the property holds.

Reply:
Right, I wonder if that might make it easier for corner cases to slip in if people thinks they can trust CI and don't run it locally with more examples.

Reply:
Rare corner cases can always happen :shrug: Supposition.jl is not generally exhaustive, but neither are hard-coded example tests. The former at least has the chance of uncovering it eventually (and reproducibly at that), while the latter gives the illusion of correctness.

I should look into making it easier to get a potential failure out of CI and onto a local machine though, maybe by having an example CI job that stores the used `ExampleDB` somewhere other than the local filesystem or having e.g. an S3-backed `ExampledDB`? That's not something you have to worry about for your PR though.

Reply:
We now have fuzz-testing as part of CI :slightly_smiling_face:

Reply:
Looks like I'll need to adjust the tests/CI to work as expected (specifically `hasparserdiverged()`)

Reply:
Ah, here we go
```
Test Summary: | Pass  Total   Time
Fuzz (5000)   |    1      1  28.8s
```

Should do `Fuzz (10000)` locally, and `Fuzz (1000)` on main or PRs that don't change `styledmarkup.jl`.